cpp/benchmarks/string/like.cpp

```diff
@@ -32,11 +45,12 @@ static void bench_like(nvbench::state& state)
   state.add_global_memory_writes<nvbench::int8_t>(n_rows);    // writes are BOOL8
 
   state.exec(nvbench::exec_tag::sync,
-             [&](nvbench::launch& launch) { auto result = cudf::strings::like(input, pattern); });
+             [&](nvbench::launch&) { cudf::strings::like(input, pattern); });
```

Comment:
I wonder if we initially had the result to avoid compiler optimizing away the whole call. I don't think it's needed, but would like to know what's the guideline for this in general.